### PR TITLE
Further verify timing for incomplete parallels fix from [JENKINS-38536]

### PR DESCRIPTION
**Pull request by svanoort**
Update plugin pom & dependencies, plus further verifies coverage of [JENKINS-38536](https://issues.jenkins-ci.org/browse/JENKINS-38536) fix (to insure we don't cause future regressions).

Downstream of https://github.com/jenkinsci/workflow-api-plugin/pull/39

### CODE REVIEW

**Comment by svanoort** [comment]
@reviewbybees

File: pom.xml [hunk 1]
```diff
@@ -26,33 +27,41 @@
...
             <artifactId>workflow-job</artifactId>
-            <version>${workflow.version}</version>
+            <version>2.0</version>
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-api</artifactId>
-            <version>2.2</version> <!-- allows consuming the new APIs -->
+            <!-- Only until we can cut a release -->
+            <version>2.17-20170605.184432-2</version>
```
Comment (jglick) [inline pom.xml]: Recommend linking to the upstream PR in a comment.

File: pom.xml [hunk 2]
```diff
@@ -6,7 +6,8 @@
     <parent>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>plugin</artifactId>
-        <version>2.9</version>
+        <version>2.29</version>
```
Comment (jglick) [inline pom.xml]: Not 2.30?

File: src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTiming.java [hunk 3]
```diff
@@ -428,10 +431,16 @@ private int parseIota(FlowNode node) {
         System.out.println("Action format: ");
         System.out.println("\t- actionClassName actionDisplayName");
         System.out.println("------------------------------------------------------------------------------------------");
+        Function<FlowNode, String> flowNodeToId = new Function<FlowNode, String>(){
+            @Override
+            public String apply(@Nullable FlowNode input) {
+                return (input != null) ? input.getId() : null;
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTiming.java]: Why are you checking for null? `FlowNode.getParents()` may not return null elements.

File: src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTiming.java [hunk 4]
```diff
@@ -428,10 +431,16 @@ private int parseIota(FlowNode node) {
...
+            }
+        };
         for (FlowNode node : sorted) {
             StringBuilder formatted = new StringBuilder();
             formatted.append('[').append(node.getId()).append(']');
-            formatted.append('{').append(StringUtils.join(node.getParentIds(), ',')).append('}');
+            formatted.append('{').append(StringUtils.join(Collections2.transform(node.getParents(), flowNodeToId), ',')).append('}');
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTiming.java]: What is the purpose of this change? Seems to be a more complicated way of writing the same thing.

File: src/test/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTimingTest.java [hunk 5]
```diff
@@ -316,6 +319,42 @@ public void testInProgress() throws Exception {
...
+                "stage 'first'\n" +
+                "parallel 'long' : { sleep 10; }, \n" +
+                "         'short': { sleep 2; }";
+
+        // This must be amateur science fiction because the exposition for the setting goes on FOREVER
+        ForkScanner scan = new ForkScanner();
+        WorkflowJob job = j.jenkins.createProject(WorkflowJob.class, "parallelTimes");
+        job.setDefinition(new CpsFlowDefinition(jobScript));
+        WorkflowRun run = job.scheduleBuild2(0).getStartCondition().get();
+        Thread.sleep(4000);  // Such a dirty hack, but this isn't reproducible with normal SemaphoreStep use
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTimingTest.java]: Because…?

File: src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTiming.java (hunk 4 above)
Comment (svanoort) [inline src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTiming.java]: getParentIds was deprecated and this update triggered enforcement of the rule.

File: pom.xml [hunk 6]
```diff
@@ -50,16 +50,15 @@
             <version>2.0</version>
         </dependency>
         <dependency>
+            <!-- https://github.com/jenkinsci/workflow-api-plugin/pull/39 -->
```
Comment (jglick) [inline pom.xml]: Well the comment is irrelevant now that you are on a release version! So delete.
